### ISSUE TITLE
Table 1 - No CLA Manager Designee Assigned - Step 2.3.1 in PRD 

### ISSUE THREAD

**Issue by pranab-bajpai** [communitybridge/easycla#1104]
## Summary
Any user associated with a company will be taken to this page after they switch to member console, select project from left navigation and EasyCLA from top navigation.

Here is the link to [wireframe ](https://wireframepro.mockflow.com/view/corporate-console#/page/Db3d69b699313a6e8106fdb78dea2b182)for the corporate console for EasyCLA.

![image](https://user-images.githubusercontent.com/57362064/81769701-ec1b1780-94a3-11ea-960e-85a4eda8b583.png)

Step 2.3.1 in Table 1 of PRD 
https://confluence.linuxfoundation.org/display/PROD/Corporate+Console+for+EasyCLA+V2

All the backend operations will be encapsulated in EasyCLA API call. See PRD for more details.

## Acceptance Criteria
- [ ] Design Changes Required ?--> Not Applicable (N/A)   
- [ ] Unit Testing Complete 
    - [ ] Unit Testing complete for backend API
    - [ ] Unit Testing complete for UI
- [ ] Functional Testing Complete 
    - [ ] Functional Testing complete for backend API
    - [ ] Functional Testing complete for UI
- [ ] Integration Testing Complete 
    - [ ] Integration Testing complete for backend API and UI
    - [ ] Integration Testing complete end to end dev deployment
- [ ] Documentation Updated 
- [ ] Any Open Sev. 1, Sev 2 bug(s)

**Comment by prasannamahajan** [comment]
Pull request:
https://github.com/communitybridge/easycla/pull/1125
https://github.com/LF-Engineering/acs-cli/pull/176

Note:
this api is accessible only to cla-manager-designee

**Comment by nirupamav** [comment]
**Tested and observed below issues. Roles permission testing is InProgress**
Tested with cladesigneedevintel/9IHQyn16jVMAsBTp!!
1. Empty response is returned when post is successful and I am not able to see the signed record in Corporate console
Below are the payloads that I tried. And I am able to post any number of times with the same data
{
"cla_group_id": "d3235128-f4ba-4127-bdc9-7160517efd99",
"company_sfid": "00117000015vpjXAAQ",
"project_sfid": "a092M00001IV3znQAD",
"send_as_email": true,
"authority_name": "clasignatorydevintel",
"authority_email": "ddeal+clasignatory+dev+intel@linuxfoundation.org"
}

{
"cla_group_id": "d3235128-f4ba-4127-bdc9-7160517efd99",
"company_sfid": "00117000015vpjXAAQ",
"project_sfid": "a092M00001IV3znQAD",
"send_as_email": false,
"return_url": "https://corporate.dev.lfcla.com/#/company/2688cdd6-404e-4a31-a5b4-51e9c81f099d"
}

{
"cla_group_id": "d3235128-f4ba-4127-bdc9-7160517efd99",
"company_sfid": "00117000015vpjXAAQ",
"project_sfid": "a092M00001IV3znQAD"
}

2. Observed post returns 200 only when clasignatory details are provided for authority and for other users always 400 
**Payload**
{
"cla_group_id": "d3235128-f4ba-4127-bdc9-7160517efd99",
"company_sfid": "00117000015vpjXAAQ",
"project_sfid": "a092M00001IV3znQAD",
"send_as_email": true,
"authority_name": "clamanager2devintel",
"authority_email": "ddeal+clamanager2+dev+intel@linuxfoundation.org"
}

**Response**
{
    "Message": "[POST /orgs/{salesforceID}/scopes][400] createOrgUsrRoleScopesBadRequest  &{Code: Message:usernames: clamanager2devintel are already associated with some organization}"
}

3. Error message for authority name is displayed even when authority name is provided in payload
![image](https://user-images.githubusercontent.com/60865903/83157942-c9366780-a121-11ea-86a1-e3dd1b29b089.png)

4. Error message for authority email is displayed even when authority email is provided in payload
![image](https://user-images.githubusercontent.com/60865903/83158097-fdaa2380-a121-11ea-96b7-aca269220771.png)

5. As per API details under Backend API column PRD, return_url is mandatory when send_as_email=false, but post is successfull with 200 response there is no validation for return_url
![image](https://user-images.githubusercontent.com/60865903/83159279-6c3bb100-a123-11ea-9794-8f1004d17a22.png)

6. authority_name accepts only spaces
![image](https://user-images.githubusercontent.com/60865903/83159661-d7858300-a123-11ea-8e33-50a741280d55.png)

**Comment by vinod-kadam** [comment]
@prasannamahajan , please update the status of this ticket? whether it is completed or still work in progress? If work in progress then please provide the ETA.

**Comment by vinod-kadam** [comment]
As updated by Prasanna,  this API will be available for integration on 4th Jun. As soon as it is ready for integration, Prasanna will update lfx team over slack channel.

**Comment by dealako** [comment]
LFX team should be unblocked now. Notified Tejas and Ahmed via Slack.

We are working on a use-case where the org service is not returning the scopes as we expect. Investigating...

**Comment by tejasadg** [comment]
@dealako @pranab-bajpai I am getting empty object for some of the project for example "project_sfid": "a092M00001IfTjSQAV",  "cla_group_id": "462cb58d-eaca-4bba-9d2c-ede401185426" As discussed with @prasannamahajan  this is the case where if the template configuration is not there for a certain project then it returns an empty object with a 200 response

**Comment by dealako** [comment]
Moved to the next sprint. Awaiting QA @nirupamav

**Comment by dealako** [comment]
@tejasadg - this is an obscure error condition.  How did this occur?

**Comment by tejasadg** [comment]
> @tejasadg - this is an obscure error condition. How did this occur?

@dealako Most of the projects in dev env does not have template configuration. That's why its return an empty object. 
cc: @prasannamahajan

**Comment by nirupamav** [comment]
Logs provided by @prasannamahajan 
https://us-east-2.console.aws.amazon.com/cloudwatch/home?region=us-east-2#logsV2:log-groups/log-group/$252Faws$252Flambda$252Fcla-backend-go-dev-api/log-events/2020$252F07$252F10$252F$255B$2524LATEST$255D2dc9b0b436a1453fa9747e7b474bfad3$3Fstart$3D2020-07-10T12$253A36$253A17.246Z

**Comment by prasannamahajan** [comment]
Added support of signing cla at foundation level
fixing bugs
@nirupamav can you open separate ticket for case 
- authority_name accepts only spaces and empty response object is returned

as error in this scenario is caused by docusign api. I couldnt find valid regex from docusign.

**Comment by nirupamav** [comment]
**_Observed that below issues are fixed_**

**1. authority user who is associated with another company. 400 with appropriate error message is returned**
![image.png](https://images.zenhubusercontent.com/5eda192d0167eac230ac5cee/3dc167b6-6daf-417a-896b-83d0665dba9c)

**2. 400 with appropriate message is returned when send_as_email=false and return_url is not provided**
![image.png](https://images.zenhubusercontent.com/5eda192d0167eac230ac5cee/59d4e24f-c671-481e-aa29-7b8d660288c3)

**4. When no data is passed to authority_name and authority_email, 400 with appropriate message is returned**
![image.png](https://images.zenhubusercontent.com/5eda192d0167eac230ac5cee/589e971e-24cc-43d6-8727-44a2f9a2d111)
![image.png](https://images.zenhubusercontent.com/5eda192d0167eac230ac5cee/b62d39b0-5569-4df1-877d-26a767579266)

**Comment by nirupamav** [comment]
Few other issues are tracked in below tickets. Since they are not blockers moving the story to QA Done
#1307, #1306, #1303, #1519
